src/endpoints/thumbnails.js

```diff
@@ -91,162 +96,213 @@ export function invalidateThumbnail(directories, type, file) {
 }
 
 /**
- * Generates a thumbnail for the given file.
- * @param {import('../users.js').UserDirectoryList} directories User directories
- * @param {ThumbnailType} type Type of the thumbnail
- * @param {string} file Name of the file
- * @returns
+ * Generates or retrieves a thumbnail for a given file.
+ * @param {import('../users.js').UserDirectoryList} directories - User's directory configuration.
+ * @param {ThumbnailType} type - Type of thumbnail ('bg', 'avatar', 'persona').
+ * @param {string} file - The filename of the image.
+ * @param {boolean} [forceGenerate=false] - Whether to force generation even if a thumbnail exists.
+ * @param {boolean|null} [isKnownAnimated=null] - If true, skips generation. If false, assumes static. If null, checks.
+ * @returns {Promise<{path: string|null, aspectRatio: number|null, resolution: number|null}>} Path to thumbnail, its aspect ratio, and resolution.
  */
-async function generateThumbnail(directories, type, file) {
-    let thumbnailFolder = getThumbnailFolder(directories, type);
-    let originalFolder = getOriginalFolder(directories, type);
+export async function generateThumbnail(directories, type, file, forceGenerate = false, isKnownAnimated = null) {
+    // If the caller has already determined the file is animated, skip processing.
+    if (isKnownAnimated === true) {
```

Comment:
No need for strict comparison with boolean literal.
```suggestion
    if (isKnownAnimated) {
```